term/src/terminalstate/image.rs

```diff
@@ -217,7 +217,7 @@ impl TerminalState {
                 xpos += x_delta;
             }
             ypos += y_delta;
-            if !params.do_not_move_cursor && y < height_in_cells - 1 {
+            if !params.do_not_move_cursor && (params.style != ImageAttachStyle::Iterm || y < height_in_cells - 1) {
```

Comment:
I feel like the logic here is hard to grok. Does this make it clearer and reflect the correct intent?

```suggestion
            let new_line = match params.style {
               ImageAttachStyle::Iterm => !params.do_not_move_cursor && y < height_in_cells - 1,
               _ => !params.do_not_move_cursor
            };
            if new_line {
               self.new_line(false);
            }
```

Reply:
As a disclaimer: this is my first time looking at Rust code (I tend to focus on C++). I'll defer to you as to whether this makes the intent clearer!